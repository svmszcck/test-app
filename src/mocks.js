import React from "react";
import configureMockStore from "redux-mock-store";
import * as redux from "react-redux";
import { NavigationContainer } from "@react-navigation/native";
import { createStackNavigator } from "@react-navigation/stack";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);
const Stack = createStackNavigator();

export const store = mockStore({
  user: {
    name: "Onur",
    avatar: null,
    favorites: [],
  },
  posts: {
    movie: {
      id: 12,
      title: "Rwer",
      poster_path: "ekjrhtkjert",
      vote_average: 5.6,
    },
  },
});

export const MockedNavigator = ({ component, params = {} }) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen
          name="MockedScreen"
          component={component}
          initialParams={params}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export const navigation = {
  navigate: jest.fn(),
};

export const mockUseSelector = jest.fn();
export const mockUseDispatch = jest.fn();

export const useDispatchSpy = jest.spyOn(redux, "useDispatch");
export const mockDispatchFn = jest.fn();
useDispatchSpy.mockReturnValue(mockDispatchFn);
