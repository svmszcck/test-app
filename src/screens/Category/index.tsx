import React, { useEffect, useMemo, useState } from "react";
import { useDispatch, useSelector } from "react-redux";

import { getMovieByGenre, resetMoviesByGenre } from "store/actions/posts";
import { Store } from "types";
import CategoryView from "./view";

const Category = ({ navigation, route }: CategoryProps) => {
  const dispatch = useDispatch();
  const [page, setPage] = useState<number>(1);
  const posts = useSelector((state: Store) => state.posts);
  const { moviesByGenre, moviesByGenreLoading } = posts;

  const id = useMemo(() => route?.params?.id, [route]);

  const name = useMemo(() => route?.params?.name, [route]);

  useEffect(() => {
    if (id) dispatch(getMovieByGenre(id, page));
    return () => {
      dispatch(resetMoviesByGenre());
    };
  }, []);

  const loadMore = () => {
    if (id) {
      dispatch(getMovieByGenre(id, page + 1));
      setPage(page + 1);
    }
  };

  return (
    <CategoryView
      navigation={navigation}
      movies={moviesByGenre}
      loading={moviesByGenreLoading}
      loadMore={loadMore}
      name={name}
    />
  );
};

type CategoryProps = {
  navigation: any;
  route: {
    params: any;
  };
};

export default Category;
