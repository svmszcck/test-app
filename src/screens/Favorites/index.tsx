import React from "react";
import { Alert } from "react-native";
import { useDispatch, useSelector } from "react-redux";

import { toggleFavorite } from "store/actions/user";
import { Store } from "types";
import FavoritesView from "./view";

const Favorites = ({ navigation }: FavoritesProps) => {
  const dispatch = useDispatch();
  const user = useSelector((state: Store) => state.user);
  const { favorites } = user;

  const removeItem = (id: number) => {
    Alert.alert(
      "Warning!",
      "Are you sure you want to remove this movie from your favorite list",
      [
        {
          text: "Cancel",
          onPress: () => null,
          style: "cancel",
        },
        { text: "Yes", onPress: () => dispatch(toggleFavorite({ id })) },
      ]
    );
  };

  return (
    <FavoritesView
      favorites={favorites}
      removeItem={removeItem}
      navigation={navigation}
    />
  );
};

type FavoritesProps = {
  navigation: any;
};

export default Favorites;
