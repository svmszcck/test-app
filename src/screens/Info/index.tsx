import React from "react";
import { View } from "react-native";

const Info = () => {
  return <View></View>;
};

export default Info;
