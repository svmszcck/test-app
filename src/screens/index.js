export { default as Home } from "./Home";
export { default as Category } from "./Category";
export { default as MovieDetails } from "./MovieDetails";
export { default as Welcome } from "./Welcome";
export { default as Profile } from "./Profile";
export { default as Favorites } from "./Favorites";
export { default as Search } from "./Search";
export { default as NotFound } from "./NotFound";
