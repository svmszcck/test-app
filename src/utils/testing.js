import "react-native";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";

const { JSDOM } = require("jsdom");

const url = "http://localhost";
const jsdom = new JSDOM("<!doctype html><html><body></body></html>", { url });
const { window } = jsdom;

jest.useFakeTimers();

export const setupEnv = () => {
  function copyProps(src, target) {
    Object.defineProperties(target, {
      ...Object.getOwnPropertyDescriptors(src),
      ...Object.getOwnPropertyDescriptors(target),
    });
  }

  global.window = window;
  global.document = window.document;
  global.navigator = {
    userAgent: "node.js",
  };
  copyProps(window, global);

  Enzyme.configure({ adapter: new Adapter() });
};
