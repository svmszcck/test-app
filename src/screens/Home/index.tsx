import React, { useEffect, useState } from "react";
import { Alert, BackHandler, Keyboard } from "react-native";
import { useDispatch, useSelector } from "react-redux";

import useBackHandler from "hooks/useBackHandler";
import { getGenres, getPopularMovies, resetPosts } from "store/actions/posts";
import { handleBackButton, showSurprise } from "utils/ui";
import { Store } from "types";
import HomeView from "./view";

const Home = ({ navigation }: HomeProps) => {
  const [categories, setCategories] = useState<boolean>(false);
  const [page, setPage] = useState<number>(1);
  const [isRefreshing, showRefresh] = useState<boolean>(false);
  const posts = useSelector((state: Store) => state.posts);
  const user = useSelector((state: Store) => state.user);
  const dispatch = useDispatch();
  const {
    genres,
    popularMovies,
    genresLoading,
    moviesLoading,
    popularMoviesLoading,
  } = posts;
  const { name } = user;

  useEffect(() => {
    dispatch(getGenres());
    dispatch(getPopularMovies(page));
  }, []);

  const loadPopularMovies = () => {
    dispatch(getPopularMovies(page + 1));
    setPage(page + 1);
  };

  const refreshPage = () => {
    showRefresh(true);
    dispatch(resetPosts());
    dispatch(getGenres());
    dispatch(getPopularMovies(1));
    setPage(1);
    showRefresh(false);
  };

  useBackHandler(() => {
    handleBackButton();
    return true;
  });

  return (
    <HomeView
      genres={genres}
      popularMovies={popularMovies}
      navigation={navigation}
      categories={categories}
      setCategories={setCategories}
      loadMore={loadPopularMovies}
      genresLoading={genresLoading}
      moviesLoading={moviesLoading}
      popularMoviesLoading={popularMoviesLoading}
      isRefreshing={isRefreshing}
      refresh={refreshPage}
      name={name}
      showMessage={() => showSurprise(name)}
    />
  );
};

type HomeProps = {
  navigation: any;
};

export default Home;
