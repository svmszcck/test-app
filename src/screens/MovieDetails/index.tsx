import React, { useEffect, useMemo } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useFocusEffect } from "@react-navigation/native";

import { getMovieDetails, resetMovie } from "store/actions/posts";
import { toggleFavorite } from "store/actions/user";
import { openLink } from "utils/general";
import { IMDB_URL } from "app_constants/api";
import { Movie, Store } from "types";
import MovieDetailsView from "./view";

const MovieDetails = ({ navigation, route }: MovieDetailsProps) => {
  const dispatch = useDispatch();
  const posts = useSelector((state: Store) => state.posts);
  const user = useSelector((state: Store) => state.user);
  const { movie, movieLoading } = posts;
  const {
    id,
    title,
    poster_path: posterPath,
    vote_average: voteAverage,
  }: Movie = movie;
  const { favorites } = user;

  const isFavorite = useMemo(
    () => favorites.find((favorite: any) => favorite?.id === id),
    [favorites, id]
  );

  useEffect(() => {
    const { id } = route.params;
    if (movie.id !== id) {
      dispatch(resetMovie());
      dispatch(getMovieDetails(id));
    }
  }, []);

  const doToggleFavorite = () => {
    dispatch(
      toggleFavorite({
        id,
        title,
        poster_path: posterPath,
        vote_average: voteAverage,
      })
    );
  };

  const openIMDB = (id: string) => openLink(`${IMDB_URL}/${id}`);

  return (
    <MovieDetailsView
      navigation={navigation}
      toggleFavorite={doToggleFavorite}
      movie={movie}
      movieLoading={movieLoading}
      isFavorite={isFavorite}
      openIMDB={openIMDB}
    />
  );
};

type MovieDetailsProps = {
  navigation: any;
  route: {
    params: any;
  };
};

export default MovieDetails;
