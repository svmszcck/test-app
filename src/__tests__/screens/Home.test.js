import React from "react";
import { Alert, TouchableOpacity } from "react-native";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Home } from "screens";
import { store, MockedNavigator } from "mocks";
import { setupEnv } from "utils/testing";

setupEnv();

describe("Testing Home Screen", () => {
  const container = (
    <Provider store={store}>
      <MockedNavigator component={Home} />
    </Provider>
  );

  it("renders correctly", () => {
    const tree = renderer.create(container).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("show welcome msg", () => {
    const wrapper = mount(container);

    const welcomeText = wrapper.find(TouchableOpacity).at(0);

    welcomeText.props().onPress();

    expect(Alert.alert).toHaveBeenCalled();
  });
});
