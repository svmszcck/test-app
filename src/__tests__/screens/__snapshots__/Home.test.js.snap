// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Screen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "padding": 20,
                            "paddingBottom": 0,
                          },
                          Object {
                            "backgroundColor": "#E4F5F3",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "marginBottom": 20,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "opacity": 1,
                            }
                          }
                        >
                          <Image
                            source={1}
                            style={
                              Object {
                                "height": 25,
                                "marginRight": 12,
                                "width": 25,
                              }
                            }
                          />
                          <Text
                            style={
                              Object {
                                "color": "#0B354F",
                                "fontWeight": "bold",
                              }
                            }
                          >
                            Hello 
                            Onur
                            , welcome :)
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "marginLeft": "auto",
                              "opacity": 1,
                              "width": 30,
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                      <RCTScrollView
                        contentContainerStyle={
                          Object {
                            "marginTop": 20,
                          }
                        }
                        refreshControl={
                          <RefreshControlMock
                            onRefresh={[Function]}
                            refreshing={false}
                          />
                        }
                        showsVerticalScrollIndicator={false}
                      >
                        <RCTRefreshControl />
                        <View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "marginBottom": 20,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#0B354F",
                                  "fontSize": 25,
                                  "marginBottom": 10,
                                }
                              }
                            >
                              Popular Movies
                            </Text>
                            <RCTScrollView
                              ListFooterComponent={
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "justifyContent": "center",
                                      "marginRight": 20,
                                      "minWidth": 150,
                                    }
                                  }
                                >
                                  <Themed.Text
                                    onPress={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "fontSize": 18,
                                          "fontWeight": "bold",
                                        },
                                        Object {
                                          "color": "#0B354F",
                                        },
                                      ]
                                    }
                                  >
                                    Load More
                                  </Themed.Text>
                                </View>
                              }
                              data={Array []}
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={true}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              showsHorizontalScrollIndicator={false}
                              stickyHeaderIndices={Array []}
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View>
                                <View
                                  onLayout={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flex": 1,
                                        "justifyContent": "center",
                                        "marginRight": 20,
                                        "minWidth": 150,
                                      }
                                    }
                                  >
                                    <Text
                                      onPress={[Function]}
                                      style={
                                        Object {
                                          "color": "#0B354F",
                                          "fontSize": 18,
                                          "fontWeight": "bold",
                                        }
                                      }
                                    >
                                      Load More
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </RCTScrollView>
                          </View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "marginBottom": 20,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#0B354F",
                                  "fontSize": 25,
                                  "marginBottom": 10,
                                }
                              }
                            >
                              Movie Categories
                            </Text>
                            <RCTScrollView
                              data={Array []}
                              disableVirtualization={false}
                              extraData={750}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={false}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEnabled={false}
                              scrollEventThrottle={50}
                              stickyHeaderIndices={Array []}
                              style={
                                Array [
                                  Object {
                                    "paddingTop": 15,
                                  },
                                  Object {
                                    "opacity": 0,
                                  },
                                ]
                              }
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View />
                            </RCTScrollView>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "padding": 20,
                            "paddingBottom": 0,
                          },
                          Object {
                            "backgroundColor": "#E4F5F3",
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "marginBottom": 20,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "row",
                              "opacity": 1,
                            }
                          }
                        >
                          <Image
                            source={1}
                            style={
                              Object {
                                "height": 25,
                                "marginRight": 12,
                                "width": 25,
                              }
                            }
                          />
                          <Text
                            style={
                              Object {
                                "color": "#0B354F",
                                "fontWeight": "bold",
                              }
                            }
                          >
                            Hello 
                            Onur
                            , welcome :)
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "marginLeft": "auto",
                              "opacity": 1,
                              "width": 30,
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                      <RCTScrollView
                        contentContainerStyle={
                          Object {
                            "marginTop": 20,
                          }
                        }
                        refreshControl={
                          <RefreshControlMock
                            onRefresh={[Function]}
                            refreshing={false}
                          />
                        }
                        showsVerticalScrollIndicator={false}
                      >
                        <RCTRefreshControl />
                        <View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "marginBottom": 20,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#0B354F",
                                  "fontSize": 25,
                                  "marginBottom": 10,
                                }
                              }
                            >
                              Popular Movies
                            </Text>
                            <RCTScrollView
                              ListFooterComponent={
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "flex": 1,
                                      "justifyContent": "center",
                                      "marginRight": 20,
                                      "minWidth": 150,
                                    }
                                  }
                                >
                                  <Themed.Text
                                    onPress={[Function]}
                                    style={
                                      Array [
                                        Object {
                                          "fontSize": 18,
                                          "fontWeight": "bold",
                                        },
                                        Object {
                                          "color": "#0B354F",
                                        },
                                      ]
                                    }
                                  >
                                    Load More
                                  </Themed.Text>
                                </View>
                              }
                              data={Array []}
                              disableVirtualization={false}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={true}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEventThrottle={50}
                              showsHorizontalScrollIndicator={false}
                              stickyHeaderIndices={Array []}
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View>
                                <View
                                  onLayout={[Function]}
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "flex": 1,
                                        "justifyContent": "center",
                                        "marginRight": 20,
                                        "minWidth": 150,
                                      }
                                    }
                                  >
                                    <Text
                                      onPress={[Function]}
                                      style={
                                        Object {
                                          "color": "#0B354F",
                                          "fontSize": 18,
                                          "fontWeight": "bold",
                                        }
                                      }
                                    >
                                      Load More
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </RCTScrollView>
                          </View>
                          <View
                            style={
                              Object {
                                "flex": 1,
                                "marginBottom": 20,
                              }
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "#0B354F",
                                  "fontSize": 25,
                                  "marginBottom": 10,
                                }
                              }
                            >
                              Movie Categories
                            </Text>
                            <RCTScrollView
                              data={Array []}
                              disableVirtualization={false}
                              extraData={750}
                              getItem={[Function]}
                              getItemCount={[Function]}
                              horizontal={false}
                              initialNumToRender={10}
                              keyExtractor={[Function]}
                              maxToRenderPerBatch={10}
                              onContentSizeChange={[Function]}
                              onEndReachedThreshold={2}
                              onLayout={[Function]}
                              onMomentumScrollEnd={[Function]}
                              onScroll={[Function]}
                              onScrollBeginDrag={[Function]}
                              onScrollEndDrag={[Function]}
                              removeClippedSubviews={false}
                              renderItem={[Function]}
                              scrollEnabled={false}
                              scrollEventThrottle={50}
                              stickyHeaderIndices={Array []}
                              style={
                                Array [
                                  Object {
                                    "paddingTop": 15,
                                  },
                                  Object {
                                    "opacity": 0,
                                  },
                                ]
                              }
                              updateCellsBatchingPeriod={50}
                              viewabilityConfigCallbackPairs={Array []}
                              windowSize={21}
                            >
                              <View />
                            </RCTScrollView>
                          </View>
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
